HotelBooking/KazexPayment.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace HotelBooking
+{
+    public class KazexPayment : IPaymentService
+    {
+        private bool _authorized = false;
```

Comment:
To consider:

That is quite clever solution, but I would like to you to have second look on this.
Your KazexPayment class is still "service", right? If this class "wrap" some service I can expect that every single time when I sent to it number and price it will just work like a charm. However that may not be a case since you have introduce private field, which adding state to service. For instance what will happen if for some reason I would like to pay in two steps hotel additional fee? Currently any other payment will end just with capture, that may not be fully correct payment.

In my opinion it will be worth to consider different approach.

Reply:
I implemented different approach, hopefully it's (slightly) better